setup.py

```diff
@@ -44,7 +41,7 @@
     extras_require["all"].extend(reqs)
 
 install_requires = [
-    "reana-commons[snakemake_reports]>=0.95.0a2,<0.96.0",
+    "reana-commons[snakemake, snakemake-xrootd]>=0.95.0a5,<0.96.0",
```

Comment:
You can remove the empty space for clarity.  (`[snakemake,snakemake-xrootd]`)

Reply:
In order to solve the report generating problem:

```console
2024-10-28 16:51:38,233 | snakemake.logging | MainThread | WARNING | Creating report...
WorkflowError:
Python package pygments must be installed to create reports.
```

We can add `pygments` manually:

```diff
diff --git a/setup.py b/setup.py
index a36c9d8..a427494 100644
--- a/setup.py
+++ b/setup.py
@@ -42,6 +42,8 @@ for key, reqs in extras_require.items():

 install_requires = [
     "reana-commons[snakemake, snakemake-xrootd]>=0.95.0a5,<0.96.0",
+    "pygments>=2.18.0"
+
 ]

 packages = find_packages()
```

This change will make all examples to pass:

```console
$ reana-dev run-example -c DEMO -w snakemake -b kubernetes --submit-only
```

P.S. I haven't checked why Snakemake installation does not add it by itself if it is required to generate reports.

P.S. Note that h4l example will need some amendments to log messages:

```console
$ rc test -w cms-h4l-snakemake-kubernetes
==> Testing file "tests/log-messages.feature"...
[ERROR] Scenario failed! Failed testcase: the engine logs should contain "reana-workflow-engine-snakemake | MainThread | INFO | scram job is finished"
[ERROR] Error log: The engine logs do not contain "reana-workflow-engine-snakemake | MainThread | INFO | scram job is finished"!
[ERROR] Scenario failed! Failed testcase: the engine logs should contain "reana-workflow-engine-snakemake | MainThread | INFO | analyze_data job is finished"
[ERROR] Error log: The engine logs do not contain "reana-workflow-engine-snakemake | MainThread | INFO | analyze_data job is finished"!
[ERROR] Scenario failed! Failed testcase: the engine logs should contain "reana-workflow-engine-snakemake | MainThread | INFO | analyze_mc job is finished"
[ERROR] Error log: The engine logs do not contain "reana-workflow-engine-snakemake | MainThread | INFO | analyze_mc job is finished"!
[ERROR] Scenario failed! Failed testcase: the engine logs should contain "reana-workflow-engine-snakemake | MainThread | INFO | make_plot job is finished"
[ERROR] Error log: The engine logs do not contain "reana-workflow-engine-snakemake | MainThread | INFO | make_plot job is finished"!
  -> SUCCESS: Scenario "The workflow start has produced the expected messages"
  -> ERROR: Scenario "The SCRAM step has produced the expected messages"
  -> ERROR: Scenario "The data analysis step has produced the expected messages"
  -> ERROR: Scenario "The montecarlo analysis step has produced the expected messages"
  -> ERROR: Scenario "The data plotting step has produced the expected messages"
  -> SUCCESS: Scenario "The workflow completion has produced the expected messages"
```

(basically remove the expected thread information in the middle)